core/src/main/scala/akka/persistence/r2dbc/query/scaladsl/R2dbcReadJournal.scala

```diff
@@ -112,7 +118,7 @@ final class R2dbcReadJournal(system: ExtendedActorSystem, config: Config, cfgPat
         metadata,
         row.entityType,
         row.slice,
-        filtered = false,
+        filtered = event.isEmpty && !row.payloadLazyLoaded,
```

Comment:
I don't think this is as intended, because we want to preserve the filtered filtered flag also for backtracking events. Fixed in 66135c2